sonar-kotlin-gradle/src/main/java/org/sonarsource/kotlin/gradle/checks/DependencyVersionHardcodedCheck.kt

```diff
@@ -0,0 +1,213 @@
+import org.jetbrains.kotlin.psi.KtLiteralStringTemplateEntry
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtStringTemplateExpression
+import org.jetbrains.kotlin.psi.KtValueArgument
+import org.jetbrains.kotlin.psi.KtVisitorVoid
+import org.jetbrains.kotlin.resolve.calls.util.getCalleeExpressionIfAny
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.checks.AbstractCheck
+import org.sonarsource.kotlin.api.frontend.KotlinFileContext
+
+private const val VERSION_ARGUMENT_INDEX = 2
+private const val VERSION_ARGUMENT_IDENTIFIER = "version"
+
+private val PATTERN_DEPENDENCY_WITH_VERSION = Regex(".+:.+:.+")
+
+private const val REF_NAME_DEPENDENCIES = "dependencies"
+private const val REF_NAME_CONSTRAINTS = "constraints"
+
+private const val REF_NAME_PROJECT_DEPENDENCY_EXT_VERSION = "version"
+
+private val REF_NAME_DEPENDENCY_HANDLER_SCOPE_EXTENSIONS by lazy {
+    setOf(
+        REF_NAME_CONSTRAINTS,
+        "annotationProcessor",
+        "antlr",
+        "api",
+        "apiElements",
+        "archives",
+        "checkstyle",
+        "codenarc",
+        "coffeeScriptBasePluginJs",
+        "compile",
+        "compileClasspath",
+        "compileOnly",
+        "default",
+        "deploy",
+        "earlib",
+        "envJsPlugin",
+        "findbugs",
+        "findbugsPlugins",
+        "implementation",
+        "jacocoAgent",
+        "jacocoAnt",
+        "jdepend",
+        "jsHintPlugin",
+        "play",
+        "playPlatform",
+        "playRun",
+        "playTest",
+        "pmd",
+        "providedCompile",
+        "providedRuntime",
+        "rhinoPluginRhinoClasspath",
+        "runtime",
+        "runtimeClasspath",
+        "runtimeElements",
+        "runtimeOnly",
+        "signatures",
+        "testAnnotationProcessor",
+        "testCompile",
+        "testCompileClasspath",
+        "testCompileOnly",
+        "testImplementation",
+        "testRuntime",
+        "testRuntimeClasspath",
+        "testRuntimeOnly",
+        "zinc",
+    )
+}
+
+private val REF_NAME_EXTERNAL_MODULE_DEPENDENCY_EXTENSIONS by lazy {
+    setOf(
+        "strictly",
+        "require",
+        "prefer",
+        "reject",
+    )
+}
+
+@Rule(key = "S6624")
+class DependencyVersionHardcodedCheck : AbstractCheck() {
+
+    override fun visitCallExpression(expression: KtCallExpression, kotlinFileContext: KotlinFileContext) {
+        if (getFunctionNameOrNull(expression) != REF_NAME_DEPENDENCIES) return
+        checkDependencyHandlerScopeLambda(expression, kotlinFileContext)
+    }
+
+    private fun checkDependencyHandlerScopeLambda(expression: KtCallExpression, kotlinFileContext: KotlinFileContext) {
+        getLambdaBlock(expression)?.acceptChildren(DependencyHandlerScopeLambdaCheck(kotlinFileContext))
+    }
+
+    private fun reportIssue(kotlinFileContext: KotlinFileContext, psiElement: PsiElement) {
+        kotlinFileContext.reportIssue(
+            psiElement,
```

Comment:
I'd prefer if we can report on the version part of a String, not on entire element

Reply:
Fixed.